src/main/java/org/opensearch/knn/index/codec/KNN990Codec/KNNQuantizationStateReader.java

```diff
@@ -0,0 +1,152 @@
+
+            List<Integer> fieldNumbers = new ArrayList<>();
+            List<Long> positions = new ArrayList<>();
+            List<Integer> lengths = new ArrayList<>();
+
+            // Read each field's metadata from the index section
+            for (int i = 0; i < numFields; i++) {
+                fieldNumbers.add(input.readInt());
+                int length = input.readInt();
+                lengths.add(length);
+                long position = input.readVLong();
+                positions.add(position);
+            }
+            // Read each field's bytes
+            for (int i = 0; i < numFields; i++) {
+                input.seek(positions.get(i));
+                byte[] stateBytes = new byte[lengths.get(i)];
+                input.readBytes(stateBytes, 0, lengths.get(i));
+                String fieldName = state.fieldInfos.fieldInfo(fieldNumbers.get(i)).getName();
+                readQuantizationStateInfos.put(fieldName, stateBytes);
+            }
+        }
+        return readQuantizationStateInfos;
+    }
+
+    /**
+     * Reads an individual quantization state for a given field
+     * @param readConfig a config class that contains necessary information for reading the state
+     * @return quantization state
+     */
+    public static QuantizationState read(QuantizationStateReadConfig readConfig) throws IOException {
+        String quantizationStateFileName = IndexFileNames.segmentFileName(
+            readConfig.getSegmentName(),
+            readConfig.getSegmentSuffix(),
+            KNNConstants.QUANTIZATION_STATE_FILE_SUFFIX
+        );
+        int fieldNumber = readConfig.getFieldInfo().getFieldNumber();
+
+        try (IndexInput input = readConfig.getDirectory().openInput(quantizationStateFileName, IOContext.READ)) {
+            CodecUtil.retrieveChecksum(input);
+            int numFields = getNumFields(input);
+
+            long position = -1;
+            int length = 0;
+
+            // Read each field's metadata from the index section, break when correct field is found
+            for (int i = 0; i < numFields; i++) {
+                int tempFieldNumber = input.readInt();
+                int tempLength = input.readInt();
+                long tempPosition = input.readVLong();
+                if (tempFieldNumber == fieldNumber) {
+                    position = tempPosition;
+                    length = tempLength;
+                    break;
+                }
+            }
+
+            if (position == -1 || length == 0) {
+                throw new IllegalArgumentException(String.format("Field %s not found", readConfig.getFieldInfo().getName()));
+            }
+
+            input.seek(position);
+            byte[] stateBytes = new byte[length];
+            input.readBytes(stateBytes, 0, length);
+            // Deserialize the byte array to a quantization state object
+            ScalarQuantizationType scalarQuantizationType = ScalarQuantizationType.fromId(
```

Comment:
This not gona work , and we don't need this plus here we need to make work of other types of Quantization Params.